### PR TITLE
Initial breakout of scheduling e2es to help assist in assignment and refactoring

### PR DESCRIPTION
**Pull request by timothysc**
**What this PR does / why we need it**:
This PR segregates the scheduling specific e2es to isolate the library which will assist both in refactoring but also auto-assignment of issues.  

**Which issue this PR fixes** 
xref: https://github.com/kubernetes/kubernetes/issues/42691#issuecomment-285563265

**Special notes for your reviewer**:
All this change does is shuffle code around and quarantine.  Behavioral, and other cleanup changes, will be in follow on PRs.  As of today, the e2es are a monolith and there is massive symbol pollution, this 1st step allows us to segregate the e2es and tease apart the dependency mess. 

**Release note**:

```
NONE
```

/cc @kubernetes/sig-scheduling-pr-reviews @kubernetes/sig-testing-pr-reviews @marun @skriss 

/cc @gmarek - same trick for load + density, etc.

### CODE REVIEW

File: test/e2e/scheduling/scheduler_predicates.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package e2e
+package scheduling
```
Comment (jbeda) [inline test/e2e/scheduling/scheduler_predicates.go]: This file could be renamed `test/e2e/scheduling/predicates.go` to save some stuttering.

File: test/e2e/scheduling/rescheduler.go [hunk 2]
```diff
@@ -109,3 +113,37 @@ func podRunningOrUnschedulable(pod *v1.Pod) bool {
...
+		Namespace:      f.Namespace.Name,
+		Timeout:        defaultTimeout,
+		Image:          framework.GetPauseImageName(f.ClientSet),
+		Replicas:       replicas,
+		CpuRequest:     request,
+	}
+	framework.ExpectNoError(framework.RunRC(*config))
+}
+
+func CreateHostPortPods(f *framework.Framework, id string, replicas int, expectRunning bool) {
```
Comment (jbeda) [inline test/e2e/scheduling/rescheduler.go]: This looks duplicated from `test/e2e/cluster_size_autoscaling.go`.  Move into util?

**Comment by timothysc** [comment]
Addressed comments.  There definitely needs to be further rationalization of the symbols into logical libraries that make more sense as follow-ons to this.

**Review by bsalamat — APPROVED** [review]
Looks good. Thanks!

/lgtm

**Comment by ConnorDoyle** [comment]
This is a great start, also LGTM

**Comment by ConnorDoyle** [comment]
@k8s-bot gce etcd3 e2e test this
@k8s-bot kops aws e2e test this

^ look like flakes (waiting for nodes timed out, gc, kubectl)

**Comment by ConnorDoyle** [comment]
@k8s-bot kops aws e2e test this

**Comment by timothysc** [comment]
@k8s-bot kops aws e2e test this

**Comment by timothysc** [comment]
One more change follows onto this one for 1.6 release.

**Comment by timothysc** [comment]
@kubernetes/test-infra-maintainers ^ kops looks like it's just failing.

**Comment by timothysc** [comment]
@k8s-bot kops aws e2e test this

**Comment by justinsb** [comment]
@timothysc kops fix has worked its way through the pipeline it looks like :-)

**Comment by timothysc** [comment]
double lgtm and we are the owners

**Comment by timothysc** [comment]
Review status: 0 of 13 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/e2e/scheduling/predicates.go, line 17 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/42968#-Kf3XJ5tJZwslDtgFdpa-r1-17:-KfDXBR5Qxi5jgySI7hq:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/3c7835b8321b30869ab707c4bbc27c19c173b0e4/test/e2e/scheduling/predicates.go#L17)):*
<details><summary><i>Previously, jbeda (Joe Beda) wrote…</i></summary><blockquote>

This file could be renamed `test/e2e/scheduling/predicates.go` to save some stuttering.
</blockquote></details>

Done.

---

*[test/e2e/scheduling/rescheduler.go, line 133 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/42968#-Kf3XJ5tJZwslDtgFdp_-r1-133:-KfDX8wcWyEIxpWdQ47r:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/3c7835b8321b30869ab707c4bbc27c19c173b0e4/test/e2e/scheduling/rescheduler.go#L133)):*
<details><summary><i>Previously, jbeda (Joe Beda) wrote…</i></summary><blockquote>

This looks duplicated from `test/e2e/cluster_size_autoscaling.go`.  Move into util?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/42968)*

**Comment by jbeda** [comment]
Reviewed 8 of 17 files at r1, 9 of 9 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/42968)*

**Comment by davidopp** [comment]
Are these tests actually running? Do we need to update anything in the Jenkins config to run them, or will it somehow pick them up in the new location automatically?

**Comment by jayunit100** [comment]
They should still be constructed into the same e2e.test binary as before , so I assume so .

**Comment by timothysc** [comment]
@davidopp Yes all expressions are being evaluated and all the tests exist in the binary.

File: test/e2e/e2e_test.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 	"testing"
 
 	"k8s.io/kubernetes/test/e2e/framework"
+	_ "k8s.io/kubernetes/test/e2e/scheduling"
```
Comment (timothysc) [inline test/e2e/e2e_test.go]: @davidopp this is the key that makes it all work, it's binary smashing the scheduling tests into it.

**Comment by timothysc** [comment]
@davidopp @kubernetes/test-infra-maintainers I see all the tests there when I run 
`e2e.test --ginkgo.dryRun=true --ginkgo.focus="SchedulerPredicates|Rescheduler"` are you seeing something in the test output post this change?

**Comment by timothysc** [comment]
Ok, I'm going to exit the fire drill I see them being run in the 24hr test grid: https://storage.googleapis.com/kubernetes-test-history/static/suite-ci-kubernetes-e2e-gce-gci-ci-serial-master.html

**Comment by bsalamat** [comment]
+1 to what @timothysc said. I also tried the e2e.go yesterday and after Tim's changes and they worked with the same exact parameters (ginkgo.focus) as I was using before the changes. Looks like the scripts pick up all the tests under test/e2e automatically.

**Pull request merged**